### ISSUE TITLE
Availbility of resource constants not properly documented

### ISSUE THREAD

**Issue by solsword** [python/cpython#116218]
# Documentation

In the [docs for the 'resource' module](https://docs.python.org/3.11/library/resource.html), RLIMIT_VMEM does not have any availability notes, however, on my system (Python 3.11.7 on ArchLinux virtualized under WSL) RLIMIT_VMEM is not defined. I see:

```sh
Python 3.11.7 (main, Jan 29 2024, 16:03:57) [GCC 13.2.1 20230801] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource as r
>>> dir(r)
['RLIMIT_AS', 'RLIMIT_CORE', 'RLIMIT_CPU', 'RLIMIT_DATA', 'RLIMIT_FSIZE', 'RLIMIT_MEMLOCK', 'RLIMIT_MSGQUEUE', 'RLIMIT_NICE', 'RLIMIT_NOFILE', 'RLIMIT_NPROC', 'RLIMIT_OFILE', 'RLIMIT_RSS', 'RLIMIT_RTPRIO', 'RLIMIT_RTTIME', 'RLIMIT_SIGPENDING', 'RLIMIT_STACK', 'RLIM_INFINITY', 'RUSAGE_CHILDREN', 'RUSAGE_SELF', 'RUSAGE_THREAD', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'error', 'getpagesize', 'getrlimit', 'getrusage', 'prlimit', 'setrlimit', 'struct_rusage']
```

Not sure which systems RLIMIT_VMEM is available on, but that should probably be mentioned.

### Linked PRs
* gh-116256
* gh-116533
* gh-116534

**Comment by Privat33r-dev** [comment]
https://bugs.python.org/issue40772

From what I could find, it's available only in [FreeBSD](https://man.freebsd.org/cgi/man.cgi?query=getrlimit&sektion=2).

**Comment by erlend-aasland** [comment]
@Privat33r-dev I think we should start with adding availability annotations to those constants in the docs.

**Comment by solsword** [comment]
My bad! I did a search for RLIMIT_VMEM on this tracker before posting, but I guess that wasn't thorough enough! (okay I just verified the default search includes "is:open" which was the problem. Indeed I didn't read the disclaimer mentioned in the other report carefully enough, but there are other constants that have explicit availability notes about FreeBSD, so it seems like this one should have that too? If it's useful I can probably figure out submitting a pull request for that label?

**Comment by Privat33r-dev** [comment]
> My bad! I did a search for RLIMIT_VMEM on this tracker before posting, but I guess that wasn't thorough enough!
No problem, we are all humans and it's absolutely normal, I have 2 issues that are irrelevant :)

> If it's useful I can probably figure out submitting a pull request for that label?
Sure, why not. Contributions are welcome.

A few useful links for you:
[General advice about contributing to Docs](https://devguide.python.org/documentation/help-documenting/index.html#helping-with-documentation)
[How to build and test docs}(https://devguide.python.org/documentation/start-documenting/#building-doc)

**Comment by Privat33r-dev** [comment]
> @Privat33r-dev I think we should start with adding availability annotations to those constants in the docs.

I made a little research. It appears that most of the `availability` params are taken from man(7) pages, while at the same time they don't cover some constants. Turns out, they are even deeper than [linux repo]() on Github posted by Linus, as some constants' history (e.g. [RLIMIT_AS](https://github.com/mpe/linux-fullhistory/blame/linux-2.4.0/include/asm-alpha/resource.h#L15)) go to version [1.3.0](https://github.com/mpe/linux-fullhistory/commit/940649fbecb9748286d0f65d57395aaea79b8e74#diff-0d856f31c554d4bc91d5cd2d530832f8b345cd7b8db78ab5e764e57ca682672d) (`include/asm-alpha/resource.h` file).

I guess that you didn't mean to check that deeply, but as a result we can add annotation to many (if not all) the constants.
